src/Service/DynamoDb/src/DynamoDbClient.php

```diff
@@ -98,7 +100,7 @@ public function batchGetItem($input): BatchGetItemOutput
             'ResourceNotFoundException' => ResourceNotFoundException::class,
             'RequestLimitExceeded' => RequestLimitExceededException::class,
             'InternalServerError' => InternalServerErrorException::class,
-        ]]));
+        ], 'usesEndpointDiscovery' => true]));
```

Comment:
does the aws cli tool actually use the discovered endpoints for dynamo api calls?

Reply:
yes, this is an opt-in option. (same for my PR)

the behavior is active when:
`data['endpointdiscovery']['required']` is true

or `data['endpointdiscovery']` exists and client is constructed with `endpointDiscoveryEnabled=true` (or associated env variable)